### PR TITLE
Refactor Pipeline REST API endpoint

### PR DESCRIPTION
**Pull request by lresende**
Move pipeline submission endpoint from
api/scheduler to api/pipeline/schedule
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: elyra/pipeline/handlers.py [hunk 1]
```diff
@@ -51,8 +56,41 @@ def post(self, *args, **kwargs):
...
+        self.log.debug("Pipeline SchedulerHandler now executing post request")
+
+        pipeline_definition = self.get_json_body()
+
+        self.log.debug("JSON payload: %s", pipeline_definition)
+
+        pipeline = PipelineParser.parse(pipeline_definition)
+
+        if pipeline.export:
+            PipelineProcessorManager.export(pipeline)
```
Comment (kevin-bates) [inline elyra/pipeline/handlers.py:82]: We should throw an exception here.  The client is calling the wrong endpoint.

The same goes for `PipelineExportHandler`.  It should confirm that `pipeline.export` is True.

File: elyra/pipeline/handlers.py [hunk 2]
```diff
@@ -51,8 +56,41 @@ def post(self, *args, **kwargs):
...
+            json_msg = json.dumps({"status": "ok",
+                                   "message": "Pipeline successfully exported"})
         else:
-            return ','.join(pipeline_array)
+            response = PipelineProcessorManager.process(pipeline)
+            json_msg = json.dumps({"status": "ok",
+                                   "message": "Pipeline successfully submitted",
```
Comment (kevin-bates) [inline elyra/pipeline/handlers.py:88]: Per this review comment https://github.com/elyra-ai/elyra/pull/560#discussion_r429986432 I don't think we should include status and dialog text in the response.  Let the UI deal with that (and it doesn't look like the UI even checks `status` or `message` from the response anyway).

File: elyra/pipeline/handlers.py [hunk 3]
```diff
@@ -51,8 +56,41 @@ def post(self, *args, **kwargs):
...
+                                   "experiment-url": response.experiment_url,
+                                   "object-storage-url": response.object_storage_url,
+                                   "object-storage-path": response.object_storage_path
+                                   })
+
+        self.set_status(200)
```
Comment (kevin-bates) [inline elyra/pipeline/handlers.py:94]: We should add a try/except block similar to the [`metadata handlers`](https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/handlers.py#L57-L64) so checked exceptions don't all get mapped to `500` codes.

These will also be reflected in the swagger doc.

File: elyra/pipeline/handlers.py [hunk 4]
```diff
@@ -51,8 +56,41 @@ def post(self, *args, **kwargs):
...
+            json_msg = json.dumps({"status": "ok",
+                                   "message": "Pipeline successfully exported"})
         else:
-            return ','.join(pipeline_array)
+            response = PipelineProcessorManager.process(pipeline)
+            json_msg = json.dumps({"status": "ok",
+                                   "message": "Pipeline successfully submitted",
+                                   "experiment-url": response.experiment_url,
+                                   "object-storage-url": response.object_storage_url,
+                                   "object-storage-path": response.object_storage_path
```
Comment (kevin-bates) [inline elyra/pipeline/handlers.py:91]: I recommend adding a `to_json()` method on `PipelineProcessorResponse`, the change this to:

```python
json_msg = json.dumps(response.to_json())
```

File: elyra/pipeline/handlers.py (hunk 2 above)
Comment (ajbozarth) [reply elyra/pipeline/handlers.py:88]: echoing my response on another comment about this, iirc this is an API "good practice" and should be set whether or not our front end uses it.

File: elyra/pipeline/handlers.py (hunk 3 above)
Comment (ajbozarth) [reply elyra/pipeline/handlers.py:94]: +1

File: elyra/pipeline/handlers.py (hunk 1 above)
Comment (lresende) [reply elyra/pipeline/handlers.py]: that has been removed from here with `rename variables and clean up` pr

**Pull request merged**